pkg/chartutil/values.go

```diff
@@ -278,6 +279,18 @@ func coalesceValues(c *chart.Chart, v map[string]interface{}) (map[string]interf
 	}
 
 	for key, val := range nv {
+		// When the YAML value is null, we remove the value's key.
+		// This allows Helm's various sources of values (value files or --set) to
+		// remove incompatible keys from any previous chart, file, or set values.
+		// ref: http://www.yaml.org/spec/1.2/spec.html#id2803362
+		var nullPattern = regexp.MustCompile("^(?:null|Null|NULL|~)$")
+		if str, ok := val.(string); ok {
```

Comment:
I don't think there is any need for a regex in here. I think the best place to do this would be down on line 297 (in this code) by inserting an `else if ok && v[key] == nil`. The YAML parser already handles all of the null cases and I tested that just to make sure.